addons/pos_self_order/controllers/mobile_entry.py

```diff
@@ -105,3 +98,14 @@ def pos_self_order_get_image(self, product_id, image_size=128, **kw):
             )
             .get_response()
         )
+
+    def get_any_pos_config_sudo(self):
+        pos_config_sudo = request.env["pos.config"].sudo().search([
+            ('|'),
+            ("self_order_view_mode", "=", True),
+            ("self_order_kiosk", "=", True)], limit=1)
+
+        if not pos_config_sudo:
+            raise werkzeug.exceptions.Unauthorized()
```

Comment:
```suggestion
            raise werkzeug.exceptions.NotFound()
```

I quickly tested and I think it's better to show NotFound page than Unauthorized.

### **Unauthorized** -- shows traceback:

<img width="1352" alt="Screenshot 2023-09-12 at 12 19 27" src="https://github.com/odoo/odoo/assets/3245568/2745c833-8d81-49f5-b860-1a50b44bb272">


### **NotFound** -- shows a more user friendly page:

<img width="1342" alt="Screenshot 2023-09-12 at 12 19 54" src="https://github.com/odoo/odoo/assets/3245568/0db076df-ba4e-41de-8cdf-d83df375f66f">